### PR TITLE
Added pgweb container on centos.

### PR DESCRIPTION
**Pull request by mohammedzee1000**

### CODE REVIEW

**Comment by mohammedzee1000** [comment]
ping @jperrin @kbsingh review and merge please.
@surajssd fyi

File: pgweb/centos7/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+FROM centos:7
+MAINTAINER Mohammed Zeeshan Ahmed <moahmed@redhat.com>
+
+RUN yum -y update
+RUN yum -y install wget unzip && yum clean all
+ENV PGWEB_VERSION 0.9.5
+
+RUN cd /tmp && \
+    wget https://github.com/sosedoff/pgweb/releases/download/v$PGWEB_VERSION/pgweb_linux_amd64.zip && \
```
Comment (kbsingh) [inline pgweb/centos7/Dockerfile]: isnt there a packaged version available somewhere ?

File: pgweb/centos7/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
+FROM centos:7
+MAINTAINER Mohammed Zeeshan Ahmed <moahmed@redhat.com>
+
+RUN yum -y update
+RUN yum -y install wget unzip && yum clean all
```
Comment (kbsingh) [inline pgweb/centos7/Dockerfile]: you likely want to merge these yum commands into 1 layer

File: pgweb/centos7/Dockerfile (hunk 1 above)
Comment (mohammedzee1000) [inline pgweb/centos7/Dockerfile]: No there isnt, i now have a machine on devcloud just for this purpose, with paas sig, epel and sclo repos installed and enabled, so i can just do something like this and find out

yum search pgweb
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: mirror.vorboss.net
 * epel: epel.check-update.co.uk
 * extras: mirror.vorboss.net
 * updates: mirror.vorboss.net
Warning: No matches found for: pgweb
No matches found

File: pgweb/centos7/Dockerfile (hunk 2 above)
Comment (mohammedzee1000) [inline pgweb/centos7/Dockerfile]: Will do

**Comment by mohammedzee1000** [comment]
@kbsingh changes applied

**Comment by mohammedzee1000** [comment]
As far as maintenance is concerned (updating the version), as of now, i will have to watch the repo from where we are getting the tarball, for newer version.I will work out a way to automate this and send it in a separate PR, as of now, this container can be pulled in.

**Comment by mohammedzee1000** [comment]
Ok, i decided to update the dockerfile to use latest release whenever it is built.

**Comment by surajssd** [comment]
@mohammedzee1000 so for upstream release tracking fedora uses https://release-monitoring.org/ aka anitya see if that can be leveraged or if anything is there in centos world that can be used.

**Comment by mohammedzee1000** [comment]
@surajssd could not find any such mechanism on centos. Still, will take this under advisement and and will see if i can figure that out. For, now i think, this should suffice for getting us a usable container.
@kbsingh comments

**Comment by mohammedzee1000** [comment]
ping @jperrin

**Comment by mohammedzee1000** [comment]
Can we have this merged @jperrin @kbsingh

**Comment by kbsingh** [comment]
-1 from my side - i still dont understand why we need a 55 line python wrapper around curl.

**Comment by mohammedzee1000** [comment]
@kbsingh, the python is for communicating with the GitHub api more easily as it gives json, tougher to process in shell to find out which is latest release and then get the download url for the same, again using GitHub api

**Comment by mohammedzee1000** [comment]
To add to that why should I install jq and unzip packages for a single run when python which already is on base image provides the function I need from each isn't that bloating the image?

File: pgweb/centos7/install-pgweb.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        if not path.exists(app_loc):
+            mkdir(app_loc)
+        with ZipFile(pgweb_downloc) as zip_file:
+            zip_file.extractall(app_loc)
+        remove(pgweb_downloc)
+        print "Creating user..."
+        execute_command(["useradd", "-ms", "/bin/bash", "pgweb"])
+        print "Done"
+        break
+if not found:
```
Comment (navidshaikh) [inline pgweb/centos7/install-pgweb.py]: any error message or warning should help here.

Comment (mohammedzee1000) [inline pgweb/centos7/install-pgweb.py]: @navidshaikh will do

**Comment by mohammedzee1000** [comment]
@kbsingh i hope i answered your question.

**Comment by jperrin** [comment]
I think it might be better to tie this to a specific version/tag that we can increment. Since this pulls the latest tarball each time the container is built, I don't see a way to track the version at all. This could result in cascading build failures if it updates beyond what dependent packages expect. 

At least going the other way, pinning to a specific version, we know exactly what the version/build failure is when it breaks.

**Comment by mohammedzee1000** [comment]
@jperrin understood, updated accordingly

**Comment by containscafeine** [comment]
@mohammedzee1000 your fixes make this a lot more optimized and sleek, but one cannot simply reuse the Dockerfile. I'd rather go with standard unix utilities and have a couple of mb's extra in my image than having a corresponding Dockerfile which requires python (language specific) know-how to iterate on.
Also, I have not seen this convention in other Dockerfiles anywhere.
I could be wrong. Thoughts?

**Comment by mohammedzee1000** [comment]
Yea ok. This PR is currenly getting messed up and  i will open new PR in a few minutes and close this once i have a reference pasted here @containscafeine

**Comment by mohammedzee1000** [comment]
Opening new PR. Will link it here

**Comment by mohammedzee1000** [comment]
New PR https://github.com/CentOS/CentOS-Dockerfiles/pull/119

**Pull request closed without merging**